### PR TITLE
feat(descriptionList): add suport to auto fit

### LINKED ISSUE #5054: Description list - add auto-fit variation
https://github.com/patternfly/patternfly/pull/3553

Adds a variation that causes the description list to lay out as many columns as will fit on the row based on a minimum width. The default minimum with with `.pf-m-auto-fit` is 250px, and is overridable by setting

`--pf-c-description-list--m-auto-fit--GridTemplateColumns--min{-on[sm, md, lg, xl, 2xl]}` - changes the min-width to the specified value at an optional breakpoint. 

You can find examples at https://pf4.patternfly.org/components/description-list

### PR DESCRIPTION
**Pull request by boaz0**
**What**: Closes #5054 

// cc @mcoker

### CODE REVIEW

File: packages/react-core/src/components/DescriptionList/examples/DescriptionList.md [hunk 1]
```diff
@@ -490,3 +490,133 @@ import PlusCircleIcon from '@patternfly/react-icons/dist/js/icons/plus-circle-ic
...
+  Button,
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListText,
+  DescriptionListGroup,
+  DescriptionListDescription
+} from '@patternfly/react-core';
+import PlusCircleIcon from '@patternfly/react-icons/dist/js/icons/plus-circle-icon';
+
+<DescriptionList isAutoFit>
```
Comment (mattnolting) [inline packages/react-core/src/components/DescriptionList/examples/DescriptionList.md:508]: `--pf-c-description-list--GridTemplateColumns--min:250px` doesn't need to bet set on the basic example, it's already set in the CSS.

**Review by mattnolting — COMMENTED** [review]
@boaz0 this is fantastic! Only saw one issue with overriding the `--pf-c-description-list--GridTemplateColumns--min` be default.

File: packages/react-core/src/components/DescriptionList/examples/DescriptionList.md (hunk 1 above)
Comment (boaz0) [reply packages/react-core/src/components/DescriptionList/examples/DescriptionList.md:508]: :+1: got it

File: packages/react-core/src/components/DescriptionList/DescriptionList.tsx [hunk 2]
```diff
@@ -22,26 +32,44 @@ export interface DescriptionListProps extends Omit<React.HTMLProps<HTMLDListElem
     xl?: '1Col' | '2Col' | '3Col';
     '2xl'?: '1Col' | '2Col' | '3Col';
   };
+  autoFitModifier?: AutoFitModifiers;
```
Comment (kmcfaul) [inline packages/react-core/src/components/DescriptionList/DescriptionList.tsx:35]: Could you add a description for this prop? The docs only show `AutoFitModifiers` as the type so it's a little unclear what should be passed in without looking at the code for the interface.

Or @redallen is it possible to get the exported interface to show on our docs page?

Comment (redallen) [reply packages/react-core/src/components/DescriptionList/DescriptionList.tsx:35]: Unfortunately not unless it's inlined. Can you inline the interface here @boaz0 ?

Comment (boaz0) [reply packages/react-core/src/components/DescriptionList/DescriptionList.tsx:35]: Sure. I forgot that the docs won't show the type well unless it's inline. :smiley_cat:

**Review by kmcfaul — DISMISSED** [review]
lgtm!

**Review by tlabaj — COMMENTED** [review]
can you update the integration app and add cypress tests.

File: packages/react-core/src/components/DescriptionList/DescriptionList.tsx [hunk 3]
```diff
@@ -22,26 +32,50 @@ export interface DescriptionListProps extends Omit<React.HTMLProps<HTMLDListElem
...
+
 export const DescriptionList: React.FunctionComponent<DescriptionListProps> = ({
   className = '',
   children = null,
   isHorizontal = false,
   isAutoColumnWidths,
+  isAutoFit,
   isInlineGrid,
   columnModifier,
+  autoFitModifier = undefined,
```
Comment (mattnolting) [inline packages/react-core/src/components/DescriptionList/DescriptionList.tsx:62]: I would probably include `Min` here, as we might need to add a `-max` value in the future.

**Review by mattnolting — COMMENTED** [review]
This is fantastic! Just one comment 🤩

File: packages/react-core/src/components/DescriptionList/DescriptionList.tsx (hunk 3 above)
Comment (boaz0) [reply packages/react-core/src/components/DescriptionList/DescriptionList.tsx]: `autoMinFitModifier` or `autoFitMinModifier` or maybe `autoFitModifierMin`? :laughing:

Comment (mattnolting) [reply packages/react-core/src/components/DescriptionList/DescriptionList.tsx]: I would say `autoFitMinModifier`.

Comment (boaz0) [reply packages/react-core/src/components/DescriptionList/DescriptionList.tsx]: Got it! :grinning:  Thanks

**Review by tlabaj — APPROVED** [review]
LGTM

**Review by kmcfaul — APPROVED** [review]

**Comment by mattnolting** [comment]
Looks like the prop is set to `auto` + `Min/Max` + `Fit` + `Modifier`. Seem like it would be `autoFit` + `Min/Max` + `Modifier`.

**Comment by boaz0** [comment]
Sorry I will fix it

**Comment by boaz0** [comment]
opened ticket #5226

**Pull request merged**